tests/token_timelock.tests.ts

```diff
@@ -0,0 +1,81 @@
+import { expect, fromSigner, setupContract } from './helpers'
+import { network } from 'redspot'
+
+const { api } = network
+
+describe('TOKEN_TIMELOCK', () => {
+    async function setup() {
+        let psp22 = await setupContract('my_psp22', 'new', '1000', 'TOKEN', 'TKN', 2)
+        const beneficiary = psp22.defaultSigner
+        let releaseTime = (Date.now() / 1000) + (24 * 60 * 60)
+        let timelock = await setupContract('my_psp22_token_timelock', 'new', psp22.contract.address, beneficiary.address, releaseTime)
+        return { psp22, timelock, beneficiary, releaseTime }
+    }
+
+    it('New works', async () => {
+        const { psp22: psp22Container, timelock: timelockContainer, beneficiary, releaseTime } = await setup()
+        const { contract: psp22 } = psp22Container
+        const { query: timelockQuery } = timelockContainer
+
+        await expect(timelockQuery.token()).to.have.output(psp22.address)
+        await expect(timelockQuery.beneficiary()).to.have.output(beneficiary.address)
+        await expect(timelockQuery.releaseTime()).to.have.output(releaseTime)
+    })
+
+    it('Release works', async () => {
+        const { psp22: psp22Container, timelock: timelockContainer, beneficiary, releaseTime } = await setup()
+        const { contract: psp22, query: psp22Query } = psp22Container
+        const { contract: timelock } = timelockContainer
+
+        let depositedTokens = 1000
+        // deposit tokens to the contract
+        await expect(fromSigner(psp22, beneficiary.address).tx.transfer(timelock.address, depositedTokens, [])).to.eventually.be.fulfilled
+        await expect(psp22Query.balanceOf(timelock.address)).to.have.output(depositedTokens)
+        await expect(psp22Query.balanceOf(beneficiary.address)).to.have.output(0)
```

Comment:
this stuff doesn't work :) see https://github.com/paritytech/subport/issues/238

Reply:
that is to `api.tx.timestamp` call

Reply:
how does it not work? The tests pass, I am not testing the block time though but if the time is not right it would return an error and fail wouldn't it?

Reply:
It indeed should pass, but api.tx.timestamp.set extrinsic is of inherent origin and can only be called by the block author. In this case, the caller is some Alice, who should not be the block author, and therefore she can't just signAndSend this stuff.
Maybe they changed how inherent extrinsic origin works? Cause idk

Reply:
oh, sorry personal acc :D

Reply:
Thanks for the explanation.

Anyway, isn't this behavior enough for a test?

Reply:
it does not work, there was a bug in the test case :smiley: